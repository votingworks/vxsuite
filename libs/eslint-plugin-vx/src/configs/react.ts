import recommended = require('./recommended');

const jsExtensions = ['.js', '.jsx'];
const tsExtensions = ['.ts', '.tsx'];
const allExtensions = jsExtensions.concat(tsExtensions);

export = {
  ...recommended,
  extends: [
    ...recommended.extends
      .map((name) => (name === 'airbnb-base' ? 'airbnb' : name))
      .filter((name) => name !== 'plugin:prettier/recommended'),
    'airbnb/hooks',
    'plugin:react/recommended',
    'plugin:jsx-a11y/recommended',
    'plugin:prettier/recommended', // Enables eslint-plugin-prettier and displays prettier errors as ESLint errors. Make sure this is always the last configuration in the extends array.
  ],
  settings: {
    ...recommended.settings,
    react: {
      version: 'detect',
    },
  },
  rules: {
    ...recommended.rules,
    'jsx-a11y/label-has-associated-control': [
      'error',
      {
        controlComponents: ['Select'],
      },
    ],
    'react/jsx-filename-extension': ['error', { extensions: allExtensions }],
    'react/jsx-fragments': ['error', 'element'],
    'react/jsx-no-bind': 'off',
    'react/jsx-no-constructed-context-values': 'off',
    'react/jsx-no-useless-fragment': ['error', { allowExpressions: true }],
    'react/jsx-one-expression-per-line': 'off',
    'react/jsx-props-no-spreading': 'off',
    'react/jsx-wrap-multilines': 'off',
    'react/prop-types': 'off',
    'react/require-default-props': 'off',
  },
};
