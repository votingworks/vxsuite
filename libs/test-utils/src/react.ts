/* eslint-disable no-console -- this file monkey-patches `console` */

/**
 * Suppresses the "Can't perform a React state update on an unmounted component" warning by
 * modifying the `console.error` method to ignore the warning.
 *
 * This is a temporary workaround until we upgrade to React 18.
 */
export function suppressReact17UnmountedWarning(): void {
  const originalConsoleError = console.error;
  console.error = (...args: unknown[]) => {
    if (
      typeof args[0] === 'string' &&
      args[0].startsWith(
        `Warning: Can't perform a React state update on an unmounted component`
      )
    ) {
      return;
    }

    originalConsoleError(...args);
  };
}
