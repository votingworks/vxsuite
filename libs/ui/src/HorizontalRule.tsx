import styled from 'styled-components'

interface Props {
  color?: string
}

export const HorizontalRule = styled.p<Props>`
  display: flex;
  align-items: center;
  margin: -0.5rem 0;
  &::after,
  &::before {
    flex: 1;
    border-top: 1px solid
      ${({ color }) => (color ? 'rgb(194, 200, 203)' : undefined)};
    content: '';
  }
  &::before {
    margin-right: 0.5rem;
  }
  &::after {
    margin-left: 0.5rem;
  }
`
