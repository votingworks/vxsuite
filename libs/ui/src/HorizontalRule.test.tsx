import React from 'react'
import { render } from '@testing-library/react'

import { HorizontalRule } from './HorizontalRule'

describe('Renders HorizontalRule', () => {
  test('with defaults', () => {
    const { container } = render(<HorizontalRule>or</HorizontalRule>)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('with custom green color', () => {
    const { container } = render(
      <HorizontalRule color="#00FF00">or</HorizontalRule>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
