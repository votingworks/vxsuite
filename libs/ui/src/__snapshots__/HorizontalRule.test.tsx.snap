// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders HorizontalRule with custom green color 1`] = `
<p
  class="sc-bdvvaa InMLu"
  color="#00FF00"
>
  or
</p>
`;

exports[`Renders HorizontalRule with defaults 1`] = `
<p
  class="sc-bdvvaa jHuSWr"
>
  or
</p>
`;
